manager/integration/tests/common.py

```diff
@@ -1937,8 +1938,37 @@ def delete_replica_processes(client, api, volname):
         exec_instance_manager(api, rm_name, delete_command)
 
 
+class replicaThread(threading.Thread):
+    """
+        This class is used for catching exception caused in threads,
+        especially for AssertionError now.
+
+        Parameters:
+            target  :       The threading function.
+            args    :       Arguments of the target fucntion.
+    """
+    def __init__(self, target, args):
+        threading.Thread.__init__(self)
+        self.target = target
+        self.args = args
+        self.asserted = False
+
+    def run(self):
+        try:
+            self.target(*self.args)
+        except AssertionError as e:
+            self.asserted = True
+            print("[DEBUG] Err: {} in the thread".format(e))
+
+    def join(self):
+        threading.Thread.join(self)
+        if self.asserted:
+            assert False
```

Comment:
Can we just save the throed assert error here? then rethrow out instead of `assert False` which is too general, and would prefer assertion w/ the correct info instead of just checking the printed logs. WDYT?

Reply:
It seems we only catch the exception at run(), but we can keep the exception and error message at run() time and reraise the exception at join() time.